src/java/org/apache/cassandra/repair/autorepair/FixedSplitTokenRangeSplitter.java

```diff
@@ -139,4 +120,34 @@ private KeyspaceRepairAssignments getRepairAssignmentsForKeyspace(AutoRepairConf
 
         return new KeyspaceRepairAssignments(priority, keyspaceName, repairAssignments);
     }
+
+    @Override
+    public void setParameter(String key, String value)
+    {
+        if (key.equals(NUMBER_OF_SUBRANGES))
+        {
+            setNumberOfSubranges(Integer.parseInt(value));
+        }
+        throw new IllegalArgumentException("Unexpected parameter '" + key + "', must be " + NUMBER_OF_SUBRANGES);
+    }
+
+    @Override
+    public Map<String, String> getParameters()
+    {
+        return Collections.unmodifiableMap(new LinkedHashMap<String, String>()
+        {{
+            put(NUMBER_OF_SUBRANGES, Integer.toString(getNumberOfSubranges()));
+        }});
```

Comment:
we can just use a singleton map, which is already immutable :
```suggestion
        return Collections.singletonMap(NUMBER_OF_SUBRANGES, Integer.toString(getNumberOfSubranges()));
```

Reply:
so much simpler, didn't realize that was available in JDK8,  will apply this in other places I used this as well.